lib/connections/service.go

```diff
@@ -995,6 +1025,7 @@ func (s *service) dialParallel(ctx context.Context, deviceID protocol.DeviceID,
 					res <- conn
 				}
 				wg.Done()
+				sema.Give(1)
```

Comment:
Better defer these, having them hanging way down here gives me the heebie-jeebies

Reply:
Also, wait, we sema.Take outside the loop but sema.Give inside?